bin/retrieve_sra_metadata.py

```diff
@@ -102,187 +111,176 @@ def download_runs_meta(run_ids, page_size=100):
         for run in runs:
             run_id = run["@accession"]
             if run_id in run_ids:
-                n_runs_found = n_runs_found + 1
-                # Write out the experiment details in JSON metadata files
-                save_ncbi_meta(experiment, sample, run)
+                num_runs_found = num_runs_found + 1
+                found_runs.append(run_id)
 
-                # Convert the data to a JSON array of controlled vocabulary terms
-                save_gemmaker_meta(experiment, sample, run)
+                # Write out the experiment details in JSON metadata files
+                save_ncbi_meta(experiment, sample, run, meta_dir)
 
-                # Write in CSV format to stdout
-                sys.stdout.write("%s,%s\n" % (run_id, exp_id))
             else:
                 sys.stderr.write('Notice: the run, %s, is part of the experiment %s but not included in the input file of runs' % (run_id, exp_id))
 
-    if n_runs_found != len(run_ids):
-        sys.stderr.write('Notice: there was an unknown problem retrieving metadata for all of the runs provided: %d != %d' % (n_runs_found, len(run_ids)))
+    if (num_runs_found != len(run_ids)):
+        # Found out which runs are missing metadata.
+        missing_runs = set(run_ids).difference(set(found_runs))
+        raise Exception ('Could not retrieve metadata for the all runs: %d != %d. Missing runs: %s' % (num_runs_found, len(run_ids), ' '.join(missing_runs)))
```

Comment:
Should we really terminate the entire workflow if a few run IDs are missing from the metadata? It seems like it would be better to let the workflow continue and try to pick up those missing metadata files later.

Reply:
That's a good question.  I think the original reasons we set it terminate rather than continue was because it allowed us to see where problems occurred that we could potentially fix and it allowed someone else to be aware if a sample wasn't going to work. Otherwise it's really easy to get through the entire thing and not realize a sample was missing.  If we adjusted the script to create a file with skipped samples and the cause then I think we can get rid of the exception and let it continue.

Reply:
Okay I will give that a shot